django/db/models/fields/__init__.py

```diff
@@ -751,7 +751,7 @@ def get_pk_value_on_save(self, instance):
         something else than None, then the returned value is used when saving
         the new instance.
         """
-        if self.default:
+        if self.default and self.default is not NOT_PROVIDED:
```

Comment:
I've got dejavu here, I feel like there was some previous discussion or work around `NOT_PROVIDED` here that I'm missing? 🤔

If this is indeed something that needs to be patched I wonder whether there's a more direct test that could be written and along with it a separate ticket? 🤔

Reply:
Perhaps
```suggestion
        if self.has_default():
```